### PR TITLE
Simplified the code of the RedirectToPreferredLocale listener

### PR DESCRIPTION
**Pull request by javiereguiluz**
Some minor changes after the merge of #199.

@voronkovich could you please review that this simplifications don't break your code? Thanks!

### CODE REVIEW

File: src/AppBundle/EventListener/RedirectToPreferredLocaleListener.php [hunk 1]
```diff
@@ -42,38 +42,33 @@ class RedirectToPreferredLocaleListener
...
         }
 
-        $this->defaultLocale = $defaultLocale;
+        $this->defaultLocale = $defaultLocale ? $this->locales[0];
 
-        if (null === $this->defaultLocale) {
-            $this->defaultLocale = $this->locales[0];
-        } elseif (!in_array($this->defaultLocale, $this->locales)) {
+        if (!in_array($this->defaultLocale, $this->locales)) {
             throw new \UnexpectedValueException(sprintf('The default locale\'s argument value "%s" must be one of "%s"', $this->defaultLocale, $locales));
```
Comment (voronkovich) [inline src/AppBundle/EventListener/RedirectToPreferredLocaleListener.php]: Missing "." at the end of the exception message.

**Comment by voronkovich** [comment]
@javiereguiluz, please remove the not needed quotes from the service definitions (I forgot to do it). See https://github.com/symfony/symfony-demo/blob/master/app/config/services.yml#L34

File: src/AppBundle/EventListener/RedirectToPreferredLocaleListener.php [hunk 2]
```diff
@@ -42,38 +42,33 @@ class RedirectToPreferredLocaleListener
...
+        $this->urlGenerator = $urlGenerator;
 
+        $this->locales = explode('|', trim($locales));
         if (empty($this->locales)) {
-            throw new \UnexpectedValueException('Locales argument value must not be empty');
+            throw new \UnexpectedValueException('The list of supported locales must not be empty.');
         }
 
-        $this->defaultLocale = $defaultLocale;
+        $this->defaultLocale = $defaultLocale ? $this->locales[0];
```
Comment (yceruto) [inline src/AppBundle/EventListener/RedirectToPreferredLocaleListener.php]: Travis says: Fatal error: syntax error, unexpected ';' in this line. Missing `:` after `?`

Comment (voronkovich) [inline src/AppBundle/EventListener/RedirectToPreferredLocaleListener.php]: @yceruto, yes, it should be looks like this:
```php
$this->defaultLocale = $defaultLocale ? $defaultLocale : $this->locales[0];
```

Comment (yceruto) [inline src/AppBundle/EventListener/RedirectToPreferredLocaleListener.php]: @voronkovich with `$this->defaultLocale = $defaultLocale ?: $this->locales[0];` work.

Comment (voronkovich) [inline src/AppBundle/EventListener/RedirectToPreferredLocaleListener.php]: @yceruto, I didn't know about this short syntax :) Thanks!

Comment (javiereguiluz) [inline src/AppBundle/EventListener/RedirectToPreferredLocaleListener.php]: @yceruto thanks for noticing me about this error. This short ternary operator works on Twig ... but not in PHP. Changed to `?:` Thanks.

**Pull request closed without merging**